### PR TITLE
Do if u want mose #86

### LINKED ISSUE #86: [Sweep] Permissions
Add live permissions to use flag setters and getters, permission to override protection flags, permission to change others zone properties

### LINKED ISSUE #98: [Bug] Incorrect changed group message
# Bug Report

## Version info

Zone Version: 1.0.0.0
Sponge API: 8.0.0
Sponge impl: Sponge Vanilla
Minecraft version: MC 1.16.5

## Description of bug

>> When a user changes group, they get the message of ``Moved Optional[<username>] from X to Y``

## How to reproduce?

>> Run the command to change a players group

### LINKED ISSUE #100: [Bug] enabled command wrong in door interaction
# Bug Report

## Version info

Zone Version: 1.0.0.0
Sponge API: 8.0.0
Sponge impl: Sponge Vanilla
Minecraft version: MC 1.16.5

## Description of bug

>> When enabling/ disabling the door interaction flag you type ``enable true/false`` however this should be ``enabled/disabled ``

## How to reproduce?

>> run the command

### LINKED ISSUE #101: [Bug] enabled command wrong in both block interaction
# Bug Report

## Version info

Zone Version: 1.0.0.0
Sponge API: 8.0.0
Sponge impl: Sponge Vanilla
Minecraft version: MC 1.16.5

## Description of bug

>> When enabling/ disabling the block interaction flags you type ``enable true/false`` however this should be ``enabled/disabled ``

## How to reproduce?

>> run the command

### LINKED ISSUE #103: [Bug] Block break flag view states group is "Optional.empty"
# Bug Report

## Version info

Zone Version: 1.0.0.0
Sponge API: 8.0.0
Sponge impl: Sponge Vanilla
Minecraft version: MC 1.16.5

## Description of bug

>> When viewing the flag info of the block break, if the group is empty, the return value states "Optional.empty", this should be "None"

## How to reproduce?

>> Run the view command for block break

### LINKED ISSUE #105: [Bug] Version info wrong
# Bug Report

## Version info

Zone Version: 1.0.0.0
Sponge API: 8.0.0
Sponge impl: Sponge Vanilla
Minecraft version: MC 1.16.5

## Description of bug

>> When viewing the info of the plugin, the version info displays as ``${version}`` but should be 1.0.0.0. 

## How to reproduce?

>> Run the command ``/zone zone plugin info``

### LINKED ISSUE #106: [Bug] Version info has wrong command
# Bug Report

## Version info

Zone Version: 1.0.0.0
Sponge API: 8.0.0
Sponge impl: Sponge Vanilla
Minecraft version: MC 1.16.5

## Description of bug

>> Version info currently has the command ``/zone zone plugin info`` this should be ``/zone plugin info`` 

## How to reproduce?

>> Run the command

### LINKED ISSUE #109: [Bug] Not all group keys are added by default
# Bug Report

## Version info

Zone Version: 1.0.0.0
Sponge API: 8.0.0
Sponge impl: Sponge Vanilla
Minecraft version: MC 1.16.5

## Description of bug

>> Not all group keys are added by default

## How to reproduce?

>> Launch zones

## Extra info

>> While this may seem like a big issue, its not as there is (or at least should be) a group command for each key required to change

### LINKED ISSUE #110: [Bug] No group key change command for commands
# Bug Report

## Version info

Zone Version: x.x.x.x
Sponge API: 8.0.0
Sponge impl: Sponge Vanilla
Minecraft version: MC 1.16.5

## Description of bug

>> Some commands do not have the ability to change the group key's group

## How to reproduce?

>> Launch the plugin

## Extra info

- [ ] Player attack prevention
- [ ] Player fall damage
- [ ] Player damage

### PR DESCRIPTION
**Pull request by Lukamon24**
Closes #86 , In progress...

### CODE REVIEW

File: src/main/java/org/zone/Permissions.java [hunk 1]
```diff
@@ -9,7 +9,25 @@
     REGION_ADMIN_INFO("zone.region.admin.info"),
     BYPASS_INTERACTION_DOOR("zone.bypass.interaction.door"),
     BYPASS_INTERACTION_BLOCK("zone.bypass.interaction.block"),
-    REGION_EDIT_BOUNDS("zone.region.edit.bounds");
+    REGION_EDIT_BOUNDS("zone.region.edit.bounds"),
+    FLAG_SET("zone.region.flag.set"),
```
Comment (mosemister) [inline src/main/java/org/zone/Permissions.java:13]: This should be less generic. Whats the flag type?

File: src/main/java/org/zone/Permissions.java [hunk 2]
```diff
@@ -9,7 +9,25 @@
...
+    REMOVE_GREETINGS("zone.region.flag.greetings.message.remove"),
+    SET_LEAVING_MESSAGE("zone.region.flag.leaving.message.set"),
+    REMOVE_LEAVING_MESSAGE("zone.region.flag.leaving.message.remove"),
+    OVERRIDE_PLAYER_FALL_DAMAGE("zone.override.fall.damage.player"),
+    OVERRIDE_PLAYER_ENTITY_DAMAGE("zone.override.entity.damage.player"),
+    OVERRIDE_MONSTER_PREVENTION("zone.override.monster.prevention"),
+    OVERRIDE_GREETINGS("zone.override.greetings"),
+    OVERRIDE_LEAVING("zone.override.leaving"),
+    OVERRIDE_PLAYER_PREVENTION("zone.override.prevention.player"),
+    BYPASS_PLAYER_PREVENTION("zone.bypass.prevention.player"),
```
Comment (mosemister) [inline src/main/java/org/zone/Permissions.java:24]: Whats the difference between override and bypass?

File: src/main/java/org/zone/Permissions.java [hunk 3]
```diff
@@ -9,7 +9,25 @@
...
+    OVERRIDE_DOOR_INTERACTION("zone.override.interaction.door"),
+    BYPASS_DOOR_INTERACTION("zone.bypass.interaction.door"),
+    OVERRIDE_ITEMFRAME_INTERACTION("zone.override.interaction.itemframe"),
+    BYPASS_ITEMFRAME_INTERACTION("zone.bypass.interaction.itemframe");
```
Comment (mosemister) [inline src/main/java/org/zone/Permissions.java:26]: The permissions need to be linked to the listener, commands, etc

File: src/main/java/org/zone/Permissions.java (hunk 1 above)
Comment (Lukamon24) [reply src/main/java/org/zone/Permissions.java:13]: eh idk this is yours?

Comment (Lukamon24) [reply src/main/java/org/zone/Permissions.java:13]: and it's a command

File: src/main/java/org/zone/Permissions.java (hunk 2 above)
Comment (Lukamon24) [reply src/main/java/org/zone/Permissions.java:24]: they r same but yes.

File: src/main/java/org/zone/Permissions.java (hunk 3 above)
Comment (Lukamon24) [reply src/main/java/org/zone/Permissions.java:26]: ikr

File: src/main/java/org/zone/Permissions.java (hunk 1 above)
Comment (mosemister) [reply src/main/java/org/zone/Permissions.java:13]: remove the flag set then ;)

File: src/main/java/org/zone/Permissions.java (hunk 2 above)
Comment (mosemister) [reply src/main/java/org/zone/Permissions.java:24]: Remove one then

**Review by mosemister — APPROVED** [review]
This is too large with a lot of serious bug fixes. Ill approve it but ill reopen the permissions sweep issue

**Pull request merged**